src/api.cc

```diff
@@ -0,0 +1,149 @@
+#include "aktualizr-lite/api.h"
+
+#include <boost/uuid/uuid_generators.hpp>
+#include <boost/uuid/uuid_io.hpp>
+
+#include "helpers.h"
+#include "libaktualizr/config.h"
+#include "liteclient.h"
+
+AkliteClient::AkliteClient(const std::vector<boost::filesystem::path>& config_dirs) {
+  Config config(config_dirs);
+  config.telemetry.report_network = !config.tls.server.empty();
+  config.telemetry.report_config = !config.tls.server.empty();
+  client_ = std::make_unique<LiteClient>(config, nullptr);
+  client_->finalizeInstall();
+}
+
+static bool compareTargets(const TufTarget& a, const TufTarget& b) { return a.Version() < b.Version(); }
+
+CheckInResult AkliteClient::CheckIn() const {
+  if (!configUploaded_) {
+    client_->reportAktualizrConfiguration();
+    configUploaded_ = true;
+  }
+  client_->reportNetworkInfo();
+  client_->reportHwInfo();
+
+  CheckInResult result(CheckInResult::Status::Ok);
+
+  Uptane::HardwareIdentifier hwidToFind(client_->config.provision.primary_ecu_hardware_id);
+
+  LOG_INFO << "Refreshing Targets metadata";
+  const auto rc = client_->updateImageMeta();
+  if (!std::get<0>(rc)) {
+    LOG_WARNING << "Unable to update latest metadata, using local copy: " << std::get<1>(rc);
+    if (!client_->checkImageMetaOffline()) {
+      LOG_ERROR << "Unable to use local copy of TUF data";
+      result.status = CheckInResult::Status::Failed;
+      return result;
+    }
+    result.status = CheckInResult::Status::OkCached;
+  }
+
+  for (const auto& t : client_->allTargets()) {
+    int ver = 0;
+    try {
+      ver = std::stoi(t.custom_version(), nullptr, 0);
+    } catch (const std::invalid_argument& exc) {
+      LOG_ERROR << "Invalid version number format: " << t.custom_version();
+      ver = -1;
+    }
+    if (!target_has_tags(t, client_->tags)) {
+      continue;
+    }
+    for (const auto& it : t.hardwareIds()) {
+      if (it == hwidToFind) {
+        result.targets_.emplace_back(t.filename(), t.sha256Hash(), ver, t.custom_data());
+        break;
+      }
+    }
+  }
+
+  std::sort(result.targets_.begin(), result.targets_.end(), compareTargets);
```

Comment:
I suppose if std::set or std::unordered_set is used with a custom comparator  then no need in a separate sort call.

Reply:
I don't have a strong opinion. I went down a bit of a rabbit hole on pros/cons for this. sounds like std::set has some downsides but I think its when you hit a much larger size of elements than we're dealing with. The other thing I was starting to think was doing one sort at the end might be faster than doing lots of sorted-inserts. 

Let me know what you think. If you want std::set - do i have to be concerned with doing any special `==` operator handling to make that container work properly?

Reply:
I think it's not really important  at this stage and may be in general taking into account a number of targets . 
The std::sort does object copying or movement (if it supports it). In this case, the `TufTarget` consists of std::string and Json::Value that supports "move", so the sort should be optimal in terms of object copying during sorting in the array.